Source/WebCore/platform/graphics/cg/ImageBufferIOSurfaceBackend.h

```diff
@@ -80,10 +82,12 @@ class WEBCORE_EXPORT ImageBufferIOSurfaceBackend : public ImageBufferCGBackend {
     unsigned bytesPerRow() const override;
 
     void finalizeDrawIntoContext(GraphicsContext& destinationContext) override;
-    void invalidateCachedNativeImage() const;
+    void invalidateCachedNativeImage();
+    void invalidateSharedNativeImage();
 
     std::unique_ptr<IOSurface> m_surface;
-    mutable bool m_mayHaveOutstandingBackingStoreReferences { false };
+    mutable bool m_mayNativeImageBeCached { false };
```

Comment:
I think this should be named more like m_mayBeUsedBy.... or m_mayBeReferencedBy... The CGImageRef references the IOSurface, rather than the IOSurface "having" an image.

Reply:
And maybe an OptionSet<> would make this more clear.

Reply:
Or maybe we turn this around and use "needs" terminology: describe what needs to happen in some future scenario.